### ISSUE TITLE
can't access lexical declaration 'n' before initialization in some patterns

### ISSUE THREAD

**Issue by rwasef1830** [trullock/NUglify#216]
The following code produces an invalid minified result (redeclaration of the variable 'n') that causes the browser to reject the script with error "can't access lexical declaration 'n' before initialization"

```js
function test() {
    const map = new Map();
    for (const [key, value] of map) alert(key + value);
}
test();
```

gets minified incorrectly to

```js
function test(){const n=new Map;for(const[n,t]of n)alert(n+t)}test()
```

Which is invalid.

@trullock

**Comment by trullock** [comment]
Before I investigate this, are you on the latest NUglify?

**Comment by rwasef1830** [comment]
@trullock yes tested on v1.11.5 in my own fork of bundler minifier

**Comment by trullock** [comment]
ok

**Comment by rwasef1830** [comment]
To clarify, the issue is that NUglify reused the same variable name "n" for the const in the outer scope and one of the loop variables.

**Comment by trullock** [comment]
Yes thanks. This is related to #79 which was hacked fixed. I'll fix this and #79 together here properly

**Comment by trullock** [comment]
Ughhhhh what a bastard of a bug to track down!

Right, fixed this and #79 properly.

see PR
